Quaternion.mpl

```diff
@@ -192,3 +195,28 @@ axisAngleQuaternion: [
     ah cos
   ) quaternion
 ];
+
+rotationQuaternion: [
+  rotation:;
+
+  rotation 1.0e-12 rotationQuaternion
+];
+
+rotationQuaternion: [rotation: threshold:;; TRUE] [
+  rotation: threshold:;;
+
+  threshold: threshold 0 rotation @ cast;
+  rotation squaredLength threshold sqr < [
+    0 rotation @ identityQuaternion
+  ] [
+    length: rotation length;
+    halfAngle: length 2.0 length cast /;
+    coefficient: halfAngle sin length /;
+    (
+      0 rotation @ coefficient *
+      1 rotation @ coefficient *
+      2 rotation @ coefficient *
+      halfAngle cos
+    ) quaternion
```

Comment:
This part is done by the `axisAngleQuaternion` above. I suggest reusing the code from this function.